### ISSUE TITLE
Message Component UX

### ISSUE THREAD

**Issue by NilsJacobsen** [opral/inlang-message-ui-components#55]
This issue tracks the progress of message component UX

**Comment by NilsJacobsen** [comment]
## Progressive disclosure is the Mandra here:

Loom: [https://www.loom.com/share/cdb1e26590784247813fb60c782e94bb](https://www.loom.com/share/cdb1e26590784247813fb60c782e94bb)<br><br>**TL:DR**

We need to design for this levels of complexity without making the most use-cases more complex! If you are not interested in my thoughts around it, please skip the Loom.

<img src="https://uploads.linear.app/4cb6d418-86ee-4425-b5c8-f91266b0ab28/f80187da-c2a0-4e13-a020-6808e67c2673/1726b0a3-4e2b-42a0-8bd4-4fe69e3cf83e?signature=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXRoIjoiLzRjYjZkNDE4LTg2ZWUtNDQyNS1iNWM4LWY5MTI2NmIwYWIyOC9mODAxODdkYS1jMmEwLTRlMTMtYTAyMC02ODA4ZTY3YzI2NzMvMTcyNmIwYTMtNGUyYi00MmEwLThiZDQtNGZlNjllM2NmODNlIiwiaWF0IjoxNzE0OTkxODAxLCJleHAiOjMzMjg1NTUxODAxfQ.Ap6tDPXITrEG_cxIygd-HHoRP3QJPh9S07-BQZdKY90 " alt="image.png" width="948" height="1070" />

**Comment by felixhaeberle** [comment]
@NilsJacobsen comments from the LOOM:

I totally agree! Make it complex only for the 2%- 10% of complex use cases, not the 90% of normal ones.

This editor shares some similarities with the one from Lokalise. Can you possibly point out some differences in where you think the Lokalise one doesn't perform well? Or is this even a comparison we want to make?

---

Now I'm interested in how this work in Sherlock haha – I made it to this point in 16:30min please don't loose me here

**Comment by NilsJacobsen** [comment]
## Integration in Sherlock:

[https://www.loom.com/share/2249876d521a414b9e5e92037837fd92](https://www.loom.com/share/2249876d521a414b9e5e92037837fd92)<br><br>**TL;DR:**<br>The message bundle component needs to go into a separate window, but for complexity level 1 we want to have something similar to what we have rn.

<img src="https://uploads.linear.app/4cb6d418-86ee-4425-b5c8-f91266b0ab28/5c4177f1-018a-4400-ba1e-fd358bbd5569/f91f6a34-3ba2-4c35-9f2d-3acea3a5e564?signature=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXRoIjoiLzRjYjZkNDE4LTg2ZWUtNDQyNS1iNWM4LWY5MTI2NmIwYWIyOC81YzQxNzdmMS0wMThhLTQ0MDAtYmExZS1mZDM1OGJiZDU1NjkvZjkxZjZhMzQtM2JhMi00YzM1LTlmMmQtM2FjZWEzYTVlNTY0IiwiaWF0IjoxNzE0OTk3MzU4LCJleHAiOjMzMjg1NTU3MzU4fQ.79Dqx0wA02HoH0sT2E8pdUKs8ZaXrG3JScbb7NPoqhA " alt="image.png" width="1689" height="1066" />

**Comment by NilsJacobsen** [comment]
## UX benchmark: 

Loom: [https://www.loom.com/share/063ee977353f42cbb2fd7f772665e4b1](https://www.loom.com/share/063ee977353f42cbb2fd7f772665e4b1)

**TL;DR:**

* Not a single solution supports no-code variants 
* That could get a differentiator.

<img src="https://uploads.linear.app/4cb6d418-86ee-4425-b5c8-f91266b0ab28/16d4044c-33de-4d17-b83c-9e2fb76aade8/4461d979-68ad-48e0-b04f-87b04746ce75?signature=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXRoIjoiLzRjYjZkNDE4LTg2ZWUtNDQyNS1iNWM4LWY5MTI2NmIwYWIyOC8xNmQ0MDQ0Yy0zM2RlLTRkMTctYjgzYy05ZTJmYjc2YWFkZTgvNDQ2MWQ5NzktNjhhZC00OGUwLWIwNGYtODdiMDQ3NDZjZTc1IiwiaWF0IjoxNzE1MDAwMTc3LCJleHAiOjMzMjg1NTYwMTc3fQ.UBIE6yECmaibknZiAGB2VP1y8UvLJ-RIriBoWoNPbbc " alt="image.png" width="798" height="474" />

**Comment by NilsJacobsen** [comment]
New update on the message UI prototype:<br>[https://www.loom.com/share/39187bc0676c44898ad6567183d5a262](https://www.loom.com/share/39187bc0676c44898ad6567183d5a262)<br>[https://www.figma.com/proto/Rw5qeH9tRDxc6SWAw9ay2z/Shared-Components?page-id=0%3A1&type=design&node-id=11-2584&viewport=-640%2C-309%2C0.11&t=X9uWv8NZcqOdmkKo-1&scaling=min-zoom&starting-point-node-id=7%3A2248&mode=design](https://www.figma.com/proto/Rw5qeH9tRDxc6SWAw9ay2z/Shared-Components?page-id=0%3A1&type=design&node-id=11-2584&viewport=-640%2C-309%2C0.11&t=X9uWv8NZcqOdmkKo-1&scaling=min-zoom&starting-point-node-id=7%3A2248&mode=design)

<img src="https://uploads.linear.app/4cb6d418-86ee-4425-b5c8-f91266b0ab28/7f911c87-19d4-4c22-8105-4c5b2a0c9edf/237023aa-71e8-4738-8696-3874fcae7dfa?signature=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXRoIjoiLzRjYjZkNDE4LTg2ZWUtNDQyNS1iNWM4LWY5MTI2NmIwYWIyOC83ZjkxMWM4Ny0xOWQ0LTRjMjItODEwNS00YzViMmEwYzllZGYvMjM3MDIzYWEtNzFlOC00NzM4LTg2OTYtMzg3NGZjYWU3ZGZhIiwiaWF0IjoxNzE1MTczNDYxLCJleHAiOjMzMjg1NzMzNDYxfQ.7vxIpwffE6sn1u7VdsrC9SwRC1bc_Er7axPc9cwkEII " alt="image.png" width="1363" height="544" />

<img src="https://uploads.linear.app/4cb6d418-86ee-4425-b5c8-f91266b0ab28/3fe6b90d-0f51-4b0a-b860-24b02b0efadf/d6359c10-fbfc-4ae6-ad16-e6334f735f63?signature=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXRoIjoiLzRjYjZkNDE4LTg2ZWUtNDQyNS1iNWM4LWY5MTI2NmIwYWIyOC8zZmU2YjkwZC0wZjUxLTRiMGEtYjg2MC0yNGIwMmIwZWZhZGYvZDYzNTljMTAtZmJmYy00YWU2LWFkMTYtZTYzMzRmNzM1ZjYzIiwiaWF0IjoxNzE1MTczNDYxLCJleHAiOjMzMjg1NzMzNDYxfQ.Nn-JMRgKmi6eCChPIL9EzJYuHxYEvVGNfa9RORlCstY " alt="image.png" width="1406" height="413" />

@jldec @martin.lysk1

**Comment by NilsJacobsen** [comment]
**New update:**<br>Loom -> [https://www.loom.com/share/abe845d917d24dee80679ac7208350e7](https://www.loom.com/share/abe845d917d24dee80679ac7208350e7)<br><br>**TL;DR**

* We have two selector adding use cases 
  * `Add (plural)function selector` adds all options as variants that are known bei the (plural) function
  * `Add string selector`  adds variant by variant because we don't know which strings are possible
  * Variants need to deletable

@loris.sigrist can you double check if 95% of the use cases can be done with that?

FYI @jldec @martin.lysk1

**Comment by jldec** [comment]
also mentioned in the loom - startig with single-input selectors/variants make sese to me, especially if one of them is very generic like a simple string / enum input.
